prime-dotnet-webapi/Models/Api/PaginatedList.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Prime.Models.Api
+{
+    public class PaginatedResponse<T>
+    {
+        public List<T> Results { get; set; }
+        public int Page { get; set; }
+        public int TotalPages { get; set; }
+        public int Total { get; set; }
+        public int PageSize { get; set; }
+
+        public PaginatedResponse(PaginatedList<T> list)
+        {
+            Results = list;
+            Page = list.Page;
+            TotalPages = list.TotalPages;
+            Total = list.Total;
+            PageSize = list.PageSize;
+        }
+    }
+    public class PaginatedList<T> : List<T>
+    {
+        private static int GetPageSize()
+        {
+            return 100;
+        }
+
+        public int Page { get; private set; }
+        public int TotalPages { get; private set; }
+        public int Total { get; private set; }
+        public int PageSize { get; private set; }
+
+        public PaginatedList(List<T> items, int count, int page, int pageSize)
+        {
+            Page = page;
+            TotalPages = (int)Math.Ceiling(count / (double)pageSize);
+            Total = count;
+            PageSize = pageSize;
+
+            AddRange(items);
+        }
+
+        public bool HasPreviousPage
+        {
+            get
+            {
+                return Page > 1;
+            }
+        }
+
+        public bool HasNextPage
+        {
+            get
+            {
+                return Page < TotalPages;
+            }
+        }
+
+        public PaginatedResponse<T> Response
+        {
+            get => new(this);
+        }
+
+        public static async Task<PaginatedList<T>> CreateAsync(IQueryable<T> source, int page)
+        {
+            var pageNumber = page < 0 ? 1 : page;
```

Comment:
Doesn't handle the case of 0 well
![image](https://user-images.githubusercontent.com/7586665/156214643-3c9f571f-f069-4f78-b47d-2b12ad3ce136.png)
![image](https://user-images.githubusercontent.com/7586665/156214930-f74721b9-76f0-48a9-8397-5f12553dd133.png)

Reply:
Yeah keep forgetting this isn't javascript and ?? only replaces null, will change that condition to be less than and equal.

Reply:
or I guess less than 1 reads nicer